joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -59,6 +59,47 @@ def _init_joint(self, minval, maxval, zeroval):
             joint['effort'] = 0.0
         return joint
 
+    def init_sdf(self, robot):
+        robot = robot.getElementsByTagName('model')[0]
+        # Find all non-fixed joints
+        for child in robot.childNodes:
+            if child.nodeType is child.TEXT_NODE:
+                continue
+            if child.localName == 'joint':
+                jtype = child.getAttribute('type')
+                if jtype in ('gearbox', 'revolute2', 'ball', 'screw', 'universal', 'fixed'):
+                    continue
+                name = child.getAttribute('name')
+                self.joint_list.append(name)
+
+                # joint limits
+                if jtype == 'continuous':
+                    minval = -math.pi
+                    maxval = math.pi
+                else:
+                    try:
+                        limit = child.getElementsByTagName('limit')[0]
+                        minval = float(limit.getElementsByTagName('lower')[0].firstChild.data)
+                        maxval = float(limit.getElementsByTagName('upper')[0].firstChild.data)
+                    except ValueError:
+                        self.get_logger().warn('%s limits are not valid!' % name)
```

Comment:
I think this is missing a `continue`:

```suggestion
                        self.get_logger().warn('%s limits are not valid!' % name)
                        continue
```